tests/serverpod_test_server/test_integration/test_tools/database_operations_in_endpoint_test.dart

```diff
@@ -82,4 +82,327 @@ void main() {
     },
     runMode: ServerpodRunMode.production,
   );
+
+  withServerpod(
+    'Given TestToolsEndpoint and rollbackDatabase afterEach',
+    (endpoints, session) {
+      group('when calling createSimpleDatasInsideTransactions', () {
+        setUpAll(() async {
+          await endpoints.testTools
+              .createSimpleDatasInsideTransactions(session, 123);
+        });
+
+        test("then finds SimpleDatas using the test session's transaction",
+            () async {
+          final simpleDatas = await SimpleData.db.find(session);
+          expect(simpleDatas.length, 2);
+          expect(simpleDatas[0].num, 123);
+          expect(simpleDatas[1].num, 123);
+        });
+
+        test('then should have been rolled back in the next test', () async {
+          var simpleDatas = await SimpleData.db.find(session);
+
+          expect(simpleDatas, hasLength(0));
+        });
+      });
+
+      group('when calling createSimpleDataAndThrowInsideTransaction', () {
+        setUpAll(() async {
+          try {
+            await endpoints.testTools
+                .createSimpleDataAndThrowInsideTransaction(session, 123);
```

Comment:
Question: If a transaction throws and is not catched in the endpoint? Will the exception be propagated back to the test?

I'm guessing these end up in 500 if it happens in a server.

Reply:
Yes, you are correct! This is intended as the strategy is that any exceptions that are thrown should bubble up just like when you are calling any other function.